gohelper.go

```diff
@@ -382,6 +385,24 @@ func extractCredentials(relationships string) (Credentials, error) {
 		return nil, err
 	}
 
+	// Address inconsistent JSON-encoding of `json:"query"`
+	for k, _ := range rels {
+		// Convert to map so interface keys are accessible
+		mappedRawQuery, ok := rels[k][0].RawQuery.(map[string]interface{})
+		if !ok {
+			// Handle empty array case
+			rels[k][0].Query.IsMaster = false
```

Comment:
I don't think this is accurate.  When there's an empty array, it means the master flag is irrelevant, not that it's false.  Generally that means there's only one, which may as well be master; so it would default to true.  Or, better, it would remain unset entirely to be consistent with the other services.

Reply:
@Crell The last commit keeps the other cases (empty object, empty array, and omitted query) unset, only setting if the `is_master` key is present. Comparing the test run on `query-fix` and `master`, it looks like that unset properties (i.e. `IsMaster   bool`) do default to their "zero values", which in this case is `false`.

For elasticsearch (empty object)

`master`:

```
elasticsearch:[{http dtsla3sy7euhc-master-7rqtwti elasticsearch   elasticsearch.internal  false  169.254.250.214 elasticsearch elasticsearch:5.4 9200 bwhgfsnjp7kzqlf7pd35dfg6mm.elasticsearch.service._.us-2.platformsh.site {false}}]
```

`query-fix`:

```
elasticsearch:[{http rjify4yjcwxaa-master-7rqtwti elasticsearch   elasticsearch.internal  false  169.254.49.124 elasticsearch elasticsearch:5.4 9200 j2dkzht3gs2yr66fb2brhoj4zu.elasticsearch.service._.eu-3.platformsh.site {false} map[]}]
```

Reply:
Here's the `query` omitted case (influxdb):

`master`:

```
influxdb:[{http rjify4yjcwxaa-master-7rqtwti influxdb   influxdb.internal  false  169.254.70.27 influxdb influxdb:1.3 8086 haz5rys6n2dsnjusqa54os3ii4.influxdb.service._.eu-3.platformsh.site {false}}]
```

`query-fix`:
```
influxdb:[{http rjify4yjcwxaa-master-7rqtwti influxdb   influxdb.internal  false  169.254.70.27 influxdb influxdb:1.3 8086 haz5rys6n2dsnjusqa54os3ii4.influxdb.service._.eu-3.platformsh.site {false} <nil>}] redis:[{redis rjify4yjcwxaa-master-7rqtwti redis   redis.internal  false  169.254.39.193 redis redis:5.0 6379 2wye4dawv22vhvozyec3o5hxfm.redis.service._.eu-3.platformsh.site {false} []}]
```